src/tree/branch.rs

```diff
@@ -103,15 +108,16 @@ impl<const DEPTH: usize> Deref for PoseidonBranch<DEPTH> {
     }
 }
 
-impl<const DEPTH: usize> Default for PoseidonBranch<DEPTH> {
-    fn default() -> Self {
-        let path = iter::repeat(PoseidonLevel::default())
-            .take(DEPTH + 1)
-            .collect();
-
-        Self { path }
-    }
-}
+// // Having a 'Default' branch probably makes no sense
```

Comment:
Better to remove than comment. `Default` is used mostly for test purposes

Reply:
Agreed